### PR TITLE
[WIP] More generic configuration

### PR DESCRIPTION
**Pull request by Randl**
https://github.com/tensorflow/tensorflow/issues/13851 
@drpngx

### CODE REVIEW

File: third_party/gpus/cuda_configure.bzl [hunk 1]
```diff
@@ -209,6 +213,38 @@ def _enable_cuda(repository_ctx):
...
+    repository_ctx: The repository context.
+
+  Returns:
+    A speculative real path of the cuda toolkit install directory.
+  """
+  cuda_nvcc_path = _DEFAULT_CUDA_NVCC_PATH
+  if _CUDA_NVCC_PATH in repository_ctx.os.environ:
+    cuda_nvcc_path = repository_ctx.os.environ[_CUDA_NVCC_PATH].strip()
+  if not repository_ctx.path(cuda_nvcc_path).exists:
+    auto_configure_fail("Cannot find cuda nvcc path.")
```
Comment (inc0) [inline third_party/gpus/cuda_configure.bzl]: I think better error msg would be "Path %s does not exist" % cuda_nvcc_path or something like that. This error is unclear imo

File: third_party/gpus/cuda_configure.bzl [hunk 2]
```diff
@@ -209,6 +213,38 @@ def _enable_cuda(repository_ctx):
...
+    repository_ctx: The repository context.
+
+  Returns:
+    A speculative real path of the cuda toolkit install directory.
+  """
+  cuda_include_path = _DEFAULT_CUDA_INCLUDE_PATH
+  if _CUDA_INCLUDE_PATH in repository_ctx.os.environ:
+    cuda_include_path = repository_ctx.os.environ[_CUDA_INCLUDE_PATH].strip()
+  if not repository_ctx.path(cuda_include_path).exists:
+    auto_configure_fail("Cannot find cuda include path.")
```
Comment (inc0) [inline third_party/gpus/cuda_configure.bzl]: ditto

File: third_party/gpus/cuda_configure.bzl [hunk 3]
```diff
@@ -923,14 +970,14 @@ def _create_local_cuda_repository(repository_ctx):
...
-  cuda_include_path = cuda_toolkit_path + "/include"
+  cuda_include_path = cuda_config.cuda_include_path
   genrules = [symlink_genrule_for_dir(repository_ctx,
       cuda_include_path, "cuda/include", "cuda-include")]
-  genrules.append(symlink_genrule_for_dir(repository_ctx,
-      cuda_toolkit_path + "/nvvm", "cuda/nvvm", "cuda-nvvm"))
-  genrules.append(symlink_genrule_for_dir(repository_ctx,
-      cuda_toolkit_path + "/extras/CUPTI/include",
-      "cuda/extras/CUPTI/include", "cuda-extras"))
+  #genrules.append(symlink_genrule_for_dir(repository_ctx,
```
Comment (inc0) [inline third_party/gpus/cuda_configure.bzl]: remove commented code, but I guess that's a WIP - just a reminder:)

**Review by inc0 — CHANGES_REQUESTED** [review]

File: third_party/gpus/cuda_configure.bzl (hunk 2 above)
Comment (case540) [reply third_party/gpus/cuda_configure.bzl]: +1, I would add the path you looked in to the error msg.

File: third_party/gpus/cuda_configure.bzl [hunk 4]
```diff
@@ -209,6 +213,38 @@ def _enable_cuda(repository_ctx):
...
+def _cuda_include_folder_path(repository_ctx):
+  """Finds the cuda toolkit directory.
+
+  Args:
+    repository_ctx: The repository context.
+
+  Returns:
+    A speculative real path of the cuda toolkit install directory.
+  """
+  cuda_include_path = _DEFAULT_CUDA_INCLUDE_PATH
```
Comment (case540) [inline third_party/gpus/cuda_configure.bzl]: Previously this looks like it was...

cuda_toolkit_path + "/include"

but now ...

_DEFAULT_CUDA_INCLUDE_PATH = "/usr/local/cuda"

Should this be "/usr/local/cuda/include" ?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @gunan: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by gunan** [comment]
Closing due to inactivity.

**Pull request closed without merging**